src/lib/Sympa/List.pm

```diff
@@ -692,6 +692,14 @@ sub _cache_put {
     my $type  = shift;
     my $value = shift;
 
+    my $mtime;
+    if ($type eq 'total' or $type eq 'is_list_member') {
+        $mtime = $self->_cache_read_expiry('member');
+    } else {
+        $mtime = $self->_cache_read_expiry($type);
+    }
+    $self->{_mtime}{$type} = $mtime;
```

Comment:
_cache_get() does the same as code block above and it allows undefined _mtime element.  Why this code block have to be inserted?

Reply:
You might be right, but I need to perform some tests on `_cache_get()` and `_cache_put()` subroutines. 

I checked the `t/` unit tests directory to find out if these subroutines are already part of the test coverage, but they are not. Therefore I propose to add new `List.t` file. But I'm not familiar with the way unit tests are organized in Sympa. 

Trying to run prove from sources raises many failures:
```
$ prove
t/01_Conf.t .............. ok   
t/Config_XML.t ........... 1/? 
#   Failed test 'use Sympa::Config_XML;'
#   at t/Config_XML.t line 12.
#     Tried to use 'Sympa::Config_XML'.
#     Error:  Can't locate Sympa/Config_XML.pm in @INC (@INC contains: /usr/share/MHonArc /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at (eval 16) line 2.
# BEGIN failed--compilation aborted at (eval 16) line 2.
Can't locate object method "new" via package "Sympa::Config_XML" (perhaps you forgot to load "Sympa::Config_XML"?) at t/Config_XML.t line 17, <DATA> line 4.
# Tests were run but no plan was declared and done_testing() was not seen.
t/Config_XML.t ........... Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 1/1 subtests 
t/DataSource_LDAP2.t ..... skipped: Test::Net::LDAP required
t/DatabaseManager.t ...... ok   
t/Database_LDAP.t ........ ok   
t/Language.t ............. Can't locate Sympa/Language.pm in @INC (@INC contains: t/stub /usr/share/MHonArc /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at t/Language.t line 15.
BEGIN failed--compilation aborted at t/Language.t line 15.
t/Language.t ............. Dubious, test returned 2 (wstat 512, 0x200)
...
```
@soji Could you give me some guidelines (or documentation reference) to help me run and complete the unit tests?

Thanks

Reply:
@salaun-urennes1 , Please do `make check`.

Reply:
And, you may have not installed some dependent Perl modules (such as XML::LibXML, Locale::Messages, ...).